packages/queued-request-controller/src/QueuedRequestController.test.ts

```diff
@@ -35,10 +36,43 @@ describe('QueuedRequestController', () => {
     };
 
     const controller = new QueuedRequestController(options);
-    expect(controller.state).toStrictEqual({});
+    expect(controller.state).toStrictEqual({ queuedRequestCount: 0 });
   });
 
   describe('enqueueRequest', () => {
+    it('counts a request as queued during processing', async () => {
+      const options: QueuedRequestControllerOptions = {
+        messenger: buildQueuedRequestControllerMessenger(),
+      };
+      const controller = new QueuedRequestController(options);
+
+      await controller.enqueueRequest(async () => {
+        expect(controller.state.queuedRequestCount).toBe(1);
+      });
+      expect(controller.state.queuedRequestCount).toBe(0);
```

Comment:
If the function that `enqueueRequest` takes never runs, then the assertion won't run. Is it worth it to create a promise and then wait for the promise to resolve, to ensure that the function runs?

``` suggestion
      const promiseForEnqueuedRequest = new Promise((resolve) => {
        controller.enqueueRequest(async () => {
          resolve();
        });
      });
      expect(controller.state.queuedRequestCount).toBe(0);
      await promiseForEnqueuedRequest;
      expect(controller.state.queuedRequestCount).toBe(1);
```

Alternatively, I suppose we could use `expect.assertions(2)`.

Reply:
This assertion is directly in the test though, and Jest will ensure the test function itself completes (or register it as a failed, timed-out test otherwise). So I think this should be safe against that possibility already